### PR TITLE
[shelly] Fix Gen2 auth, improved security for Gen1 auth, improved discovery

### PR DESCRIPTION
**Pull request by markus7017**
- #15399: Rework device discovery and initialization: Use /shelly to retrieve device attributes, thing discovery is now based on reported device model id rather than service name. If is not recognized the service name will be used as fallback. Even password protected things are now created with the thing type matching the model rather than shellydevice, which is still supported for backward compatibility.
- #15397: Security fix for Gen1: We don’t expose basic auth (BASE64 encoded password) anymore when not requested by the device (/shelly.auth = true). Gen2 uses digest authentication, which is not affected by this issues.
- #15392: Version detection for Gen2 devices is fixed and now also handles beta versions correctly (e.g. 1.0.0-beta6), they are displayed correctly in Shelly Manager
- Code cleanup of duplicate device attributes, which causes a bunch of dummy changes (/referred variables names have changed)

### CODE REVIEW

File: bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api/ShellyDeviceProfile.java [hunk 1]
```diff
@@ -47,32 +48,28 @@
 @NonNullByDefault
 public class ShellyDeviceProfile {
     private final Logger logger = LoggerFactory.getLogger(ShellyDeviceProfile.class);
-    private static final Pattern VERSION_PATTERN = Pattern.compile("v\\d+\\.\\d+\\.\\d+(-[a-z0-9]*)?");
+    private static final Pattern GEN1_VERSION_PATTERN = Pattern.compile("v\\d+\\.\\d+\\.\\d+(-[a-z0-9]*)?");
+    private static final Pattern GEN2_VERSION_PATTERN = Pattern.compile("\\d+\\.\\d+\\.\\d+(-[a-fh-z0-9]*)?");
```
Comment (lolodomo) [inline bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api/ShellyDeviceProfile.java:52]: Just to be sure: the last part can contain any letter in lower case except g ?

**Review by lolodomo — CHANGES_REQUESTED** [review]

File: bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api/ShellyDeviceProfile.java (hunk 1 above)
Comment (markus7017) [reply bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api/ShellyDeviceProfile.java:52]: yes, that was the best approach, they have an inconsistent naming schema, not easy to handle. <version>-pre-xy-g<build number>, so excluding g helps to strip the build number

File: bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api2/Shelly2ApiClient.java [hunk 2]
```diff
@@ -148,6 +148,11 @@ public Shelly2ApiClient(String thingName, ShellyThingConfiguration config, HttpC
         MAP_ROLLER_STATE.put(SHELLY2_RSTATE_STOPPED, SHELLY_RSTATE_STOP);
         MAP_ROLLER_STATE.put(SHELLY2_RSTATE_CALIB, SHELLY2_RSTATE_CALIB); // Gen2-only
     }
+    protected static final Map<String, String> MAP_PROFILE = new HashMap<>();
+    static {
+        MAP_PROFILE.put(SHELLY_CLASS_RELAY, SHELLY2_PROFILE_RELAY);
+        MAP_PROFILE.put(SHELLY_CLASS_ROLLER, SHELLY2_PROFILE_COVER);
+    }
```
[the remark this answers is not in the document]
Comment (markus7017) [reply bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api2/Shelly2ApiClient.java:155]: I know, this comes in the next PR for all those maps, but I tried to keep this one as small as possible

File: bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/manager/ShellyManagerConstants.java [hunk 3]
```diff
@@ -82,6 +84,7 @@ public class ShellyManagerConstants {
     public static final String ATTRIBUTE_MESSAGE = "message";
     public static final String ATTRIBUTE_TOTAL_DEV = "totalDevices";
     public static final String ATTRIBUTE_STATUS_ICON = "iconStatus";
+    public static final String ATTRIBUTE_DEVICEIP = CONFIG_DEVICEIP;
```
[the remark this answers is not in the document]
Comment (markus7017) [reply bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/manager/ShellyManagerConstants.java]: ok, removed (moved to next PR)

**Comment by markus7017** [comment]
@lolodomo ready to move forward

File: bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api2/Shelly2ApiClient.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (lolodomo) [reply bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api2/Shelly2ApiClient.java:155]: But this is new code you added, why not doing it now even if you do not change all others ?

[the remark this answers is not in the document]
Comment (lolodomo) [reply bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api2/Shelly2ApiClient.java:155]: This was a suggestion, not a request for change. I close my comment.

**Review by lolodomo — APPROVED** [review]
LGTM even if some added code could have been done in a more modern way.

**Comment by lolodomo** [comment]
@markus7017 : please use in your description the syntax "Closes #number" for an automatic closure of issues when the PR is merged.

File: bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api2/Shelly2ApiClient.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (markus7017) [reply bundles/org.openhab.binding.shelly/src/main/java/org/openhab/binding/shelly/internal/api2/Shelly2ApiClient.java:155]: ok, changed them

**Pull request merged**